### DISCUSSION TITLE
import inconsistancy?

### DISCUSSION THREAD

**Discussion by mmliam** [micropython/micropython#9751] [General]
I can code:
`from time import sleep`, or `from machine import Pin` without issues.
However when I code:
`from network import WLAN`, it throws the error "network not defined" even though it's listed in the modules the same as 'time' or 'machine'.
I need to code: `import network` to resolve the error.

The Network module list:
 
>  WLAN -- <function>
>   phy_mode -- <function>
>   STA_IF -- 0
>   AP_IF -- 1
>   STAT_IDLE -- 0
>   STAT_CONNECTING -- 1
>   STAT_WRONG_PASSWORD -- 2
>   STAT_NO_AP_FOUND -- 3
>   STAT_CONNECT_FAIL -- 4
>   STAT_GOT_IP -- 5
>   MODE_11B -- 1
>   MODE_11G -- 2
>   MODE_11N -- 3
>   AUTH_OPEN -- 0
>   AUTH_WEP -- 1
>   AUTH_WPA_PSK -- 2
>   AUTH_WPA2_PSK -- 3
>   AUTH_WPA_WPA2_PSK -- 4

Is it because I also referenced STA_IF that caused the error to refer to the entire module as being undefined?
Had I simply used the value '0" instead of STA_IF that would have also resolved the error?

MM

**Comment by robert-hh** [comment]
Please add the port and firmware version you use.

**Reply by mmliam** [reply]
Hi robert,
I think I just discovered the reason for the problem; I need to research this further.
So for now I'd like to rescind the the post.

**Reply by mmliam** [reply]
Hi robert,

I was able to resolve the problem.
I was originally using `import network`, and using the references network.WLAN & network.STF_IF
When I tried `from network import WLAN` I forgot the change network.WLAN to just WLAN, and indeed I need to use the value '0' for 'STA_IF'.

Thanks for replying
